File: app/src/main/java/com/example/farmer/mainacthandler/SearchHelper.java
package com.example.farmer.mainacthandler;


import android.widget.Toast;

import com.example.farmer.MyAdapter;

import java.util.ArrayList;
import java.util.List;

public class SearchHelper {

    public static void initializeDummyData(List<String> dataList) {
        dataList.add("Corn");
        dataList.add("Wheat");
        dataList.add("Rice");
        dataList.add("Sugarcane");
        dataList.add("Barley");
    }

    public static void performSearch(String query, List<String> dataList, MyAdapter adapter) {
        List<String> filteredList = new ArrayList<>();

        for (String item : dataList) {
            if (item.toLowerCase().contains(query.toLowerCase())) {
                filteredList.add(item);
            }
        }


    }
}

File: app/src/main/java/com/example/farmer/mainacthandler/SearchHelper.kt
package com.example.farmer.mainacthandler

import com.example.farmer.MyAdapter
import java.util.Locale


object SearchHelper {
    fun initializeDummyData(dataList: MutableList<String?>) {
        dataList.add("Corn")
        dataList.add("Wheat")
        dataList.add("Rice")
        dataList.add("Sugarcane")
        dataList.add("Barley")
    }

    fun performSearch(query: String, dataList: MutableList<String>, adapter: MyAdapter?) {
        val filteredList: MutableList<String?> = ArrayList<String?>()

        for (item in dataList) {
            if (item.lowercase(Locale.getDefault())
                    .contains(query.lowercase(Locale.getDefault()))
            ) {
                filteredList.add(item)
            }
        }
    }
}

